### PR TITLE
lets see

### PR DESCRIPTION
**Pull request by Havrysh**
Pls leave a bit more specific comment on my mistakes so i can fix it with deeper understanding, tnx

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,53 @@
...
+export enum SortOrder {
+  asc = 'asc',
+  desc = 'desc',
+}
 
+// eslint-disable-next-line no-multiple-empty-lines
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+// eslint-disable-next-line max-len
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:26]: do not reset eslint rules

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,53 @@
...
+export function sortStudents(students: Student, sortBy: SortType, order: SortOrder): [] {
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:27]:
```suggestion
export function sortStudents(
  students: Student,
  sortBy: SortType,
  order: SortOrder,
): Student[] {
```

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,53 @@
...
+  if (order === SortOrder.asc) {
+    switch (sortBy) {
+      case SortType.Name || SortType.Surname:
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:30]:
```suggestion
      case SortType.Name:
      case SortType.Surname:
```
better use such syntax

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,53 @@
...
+    }
+  } else {
+    switch (sortBy) {
+      case SortType.Name || SortType.Surname:
+        return students.slice().sort((x, y) =>
+          y.sortBy.localeCompare(x.sortBy));
+      case SortType.Age || SortType.Married:
+        return students.slice().sort((x, y) => y.sortBy - x.sortBy);
+      default:
+        return students.slice().sort((x, y) =>
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:48]:
```suggestion
        return [...students].sort((x, y) =>
```
better make copy in such way

Comment (Bohdan-Maksymiuk) [reply src/sortStudents.ts:48]: and `student1`, `student2` instead of x and y

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 5]
```diff
@@ -1,16 +1,61 @@
...
+        return [...students].sort((student1, student2) =>
+          student1.sortBy.localeCompare(student2.sortBy));
+      case SortType.Age || SortType.Married:
+        return [...students].sort((student1, student2) =>
+          student2.sortBy - student1.sortBy);
+      default:
+        [...students].sort((student1, student2) =>
+          student2.sortBy.reduce((sum, t) => sum + t, 0)
+          - student1.sortBy.reduce((sum, t) => sum + t, 0));
+    }
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts]: it is not default, what if we add some new sortBy, or someone make a typo in sortBy string?

File: src/sortStudents.ts [hunk 6]
```diff
@@ -1,16 +1,50 @@
...
+export function sortStudents(
+  students: Student,
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((student1, student2) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:32]:
```suggestion
        return order === SortOrder.Asc
```

File: src/sortStudents.ts [hunk 7]
```diff
@@ -1,16 +1,50 @@
...
+          ? student1[sortBy].localeCompare(student2[sortBy])
+          : student2[sortBy].localeCompare(student1[sortBy]);
 
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:38]: .

File: src/sortStudents.ts [hunk 8]
```diff
@@ -1,16 +1,50 @@
...
+          ? +student1[sortBy] - +student2[sortBy]
+          : +student2[sortBy] - +student1[sortBy];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      default:
+        return order === 'asc'
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:43]: .

File: src/sortStudents.ts [hunk 9]
```diff
@@ -1,16 +1,50 @@
...
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageAge'
 }
 
-// create SortOrder type
-export type SortOrder;
+export enum SortOrder {
+  asc = 'asc',
+  desc = 'desc',
```
Comment (Bohdan-Maksymiuk) [inline src/sortStudents.ts:20]:
```suggestion
  Asc = 'asc',
  Desc = 'desc',
```
capitalize for enum names

File: src/sortStudents.ts (hunk 5 above)
Comment (Bohdan-Maksymiuk) [reply src/sortStudents.ts]: this comment not fixed

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 10]
```diff
@@ -1,16 +1,52 @@
...
+      case SortType.Married:
+        return order === SortOrder.Asc
+          ? +student1[sortBy] - +student2[sortBy]
+          : +student2[sortBy] - +student1[sortBy];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      case SortType.AverageGrade:
+        return order === SortOrder.Asc
+          ? student1.sortBy.reduce((sum, t) => sum + t, 0)
```
Comment (marina-tilniak) [inline src/sortStudents.ts:44]: тут треба знайти середнє,а не суму

Comment (marina-tilniak) [reply src/sortStudents.ts:44]: student1.sortBy - такий запис не скомпілюється. У студента немає властивлсті sortBy

File: src/sortStudents.ts [hunk 11]
```diff
@@ -1,16 +1,52 @@
...
 
-// create SortOrder type
-export type SortOrder;
+export enum SortOrder {
+  Asc = 'asc',
+  Desc = 'desc',
+}
+
+export function sortStudents(
+  students: Student,
```
Comment (marina-tilniak) [inline src/sortStudents.ts:24]:
```suggestion
  students: Student[],
```

**Review by marina-tilniak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**